src/nvim/api/buffer.c

```diff
@@ -1134,67 +1134,63 @@ Boolean nvim_buf_is_valid(Buffer buffer)
   return ret;
 }
 
-/// Delete a named mark
+/// Deletes a named mark in the buffer
+//
+/// Marks are (1,0)-indexed. |api-indexing|
 ///
-/// @param buffer     Buffer handle, or 0 for current buffer
-/// @param name       Mark name
+/// @note Only deletes marks associated with the buffer
+///
+/// @param buffer     Buffer to set the mark on
+/// @param name       Mark name {a-zZ0-9}
 void nvim_buf_del_mark(Buffer buffer, String name, Error *err)
   FUNC_API_SINCE(8)
 {
-  nvim_buf_set_mark(buffer, name, 0, 0, err);
+  buf_T *buf = find_buffer_by_handle(buffer, err);
+  if (!buf) {
+    return;
+  }
+  if (name.size != 1) {
+    api_set_error(err, kErrorTypeValidation,
+                  "Mark name must be a single character");
```

Comment:
Maybe you want to exit the function here ?

Reply:
Thanks! 😄